daemon/api.go

```diff
@@ -303,16 +305,32 @@ func getSnapsInfo(c *Command, r *http.Request) Response {
 				resource = url.String()
 			}
 
-			results[name+"."+origin] = webify(m, resource)
+			fullname := name + "." + origin
+
+			// strings.Contains(fullname, "") is true
+			if !strings.Contains(fullname, searchTerm) {
+				continue
+			}
+
+			results[fullname] = webify(m, resource)
 		}
 	}
 
 	if includeStore {
-		// TODO: If there are no results (local or remote), report the error. If
-		//       there are results at all, inform that the result is partial.
-		found, _ := newRemoteRepo().All()
-		if len(found) > 0 {
+		repo := newRemoteRepo()
+		var found []snappy.Part
+
+		if searchTerm != "" {
+			found, _ = repo.Find(searchTerm)
 			sources = append(sources, "store")
```

Comment:
Yes, I've been thinking about this. I think we want sources to indicate
whether we successfully queried the source (whether it returned anything it
not), and a new warnings field that lists things life the store being
unreachable. Perhaps also a per snap sources to say where each snap info
comes from
On 27 Jan 2016 09:59, "Samuele Pedroni" <notifications@github.com> wrote:

> In daemon/api.go
> <https://github.com/ubuntu-core/snappy/pull/371#discussion_r50961979>:
>
> >  		}
> >  	}
> >
> >  	if includeStore {
> > -		// TODO: If there are no results (local or remote), report the error. If
> > -		//       there are results at all, inform that the result is partial.
> > -		found, _ := newRemoteRepo().All()
> > -		if len(found) > 0 {
> > +		repo := newRemoteRepo()
> > +		var found []snappy.Part
> > +
> > +		if searchTerm != "" {
> > +			found, _ = repo.Find(searchTerm)
> >  			sources = append(sources, "store")
>
> why is this one not conditional on the size of found? like below, or the
> one below unconditional, or should it be conditional only if sources wasn't
> given to start with and we used the fallback
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ubuntu-core/snappy/pull/371/files#r50961979>.
>